pkg/resource/alias/hooks.go

```diff
@@ -235,9 +273,273 @@ func (rm *resourceManager) setResourceAdditionalFields(
+		// so we need to handle this case.
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.ErrorCode() == "ResourceNotFoundException" {
+			ko.Spec.Permissions = []*svcapitypes.AddPermissionInput{}
+			// set err to nil so we don't log an error
+			err = nil
+			return nil
+		}
+		return err
+	}
+
+	policyDoc := &policy.Policy{}
+	err = json.Unmarshal([]byte(*output.Policy), policyDoc)
+	if err != nil {
+		return err
+	}
+
+	// Convert policy statements to permissions
+	var permissions []*svcapitypes.AddPermissionInput
+	if policyDoc.Statements != nil {
+		for _, stmt := range policyDoc.Statements.Values() {
+			if stmt.Sid == "" { // skip empty SID statements
+				continue
+			}
+
+			permission := &svcapitypes.AddPermissionInput{
+				StatementID: aws.String(stmt.Sid),
+			}
+
+			if stmt.Action != nil && len(stmt.Action.Values()) > 0 {
+				permission.Action = aws.String(stmt.Action.Values()[0])
+			}
+
+			if stmt.Principal != nil {
+				if stmt.Principal.Service() != nil && len(stmt.Principal.Service().Values()) > 0 {
+					permission.Principal = aws.String(stmt.Principal.Service().Values()[0])
+				} else if stmt.Principal.AWS() != nil && len(stmt.Principal.AWS().Values()) > 0 {
+					permission.Principal = aws.String(stmt.Principal.AWS().Values()[0])
+				}
+			}
+
+			if stmt.Condition != nil {
+				// ArnLike condition
+				if arnCond, ok := stmt.Condition["ArnLike"]; ok {
+					if sourceArn, ok := arnCond["AWS:SourceArn"]; ok {
+						strValues, _, _ := sourceArn.Values()
+						if len(strValues) > 0 {
+							permission.SourceARN = aws.String(strValues[0])
+						}
+					}
+				}
+
+				// StringEquals condition
+				if stringCond, ok := stmt.Condition["StringEquals"]; ok {
+					if sourceAcct, ok := stringCond["AWS:SourceAccount"]; ok {
+						strValues, _, _ := sourceAcct.Values()
+						if len(strValues) > 0 {
+							permission.SourceAccount = aws.String(strValues[0])
+						}
+					}
+
+					// Extract EventSourceToken
+					if token, ok := stringCond["lambda:EventSourceToken"]; ok {
+						strValues, _, _ := token.Values()
+						if len(strValues) > 0 {
+							permission.EventSourceToken = aws.String(strValues[0])
+						}
+					}
+
+					// Extract PrincipalOrgID
+					if orgID, ok := stringCond["aws:PrincipalOrgID"]; ok {
+						strValues, _, _ := orgID.Values()
+						if len(strValues) > 0 {
+							permission.PrincipalOrgID = aws.String(strValues[0])
+						}
+					}
+				}
+			}
+
+			permissions = append(permissions, permission)
+		}
+	}
+
+	ko.Spec.Permissions = permissions
 	return nil
 }
 
+// compares the desired and existing permissions and  returns two slices:
+// permissions to remove and permissions to add. Updates are represented
+// as a permission to remove and a permission to add.
+//
+// Yes the API doesn't support updating permissions directly... yikes.
+func comparePermissions(
+	desired []*svcapitypes.AddPermissionInput,
+	existing []*svcapitypes.AddPermissionInput,
```

Comment:
```suggestion
	latest []*svcapitypes.AddPermissionInput,
```

nit: can we call this latest for consistency